beetsplug/bandcamp/_search.py

```diff
@@ -26,16 +26,10 @@ def _f(field: str) -> str:
     re.compile(r"\n\s+by " + _f("artist")),
     re.compile(r"\n\s+released " + _f("date")),
     re.compile(r"\n\s+(?P<tracks>\d+) tracks"),
-    re.compile(
-        r"""
-(?P<url>
-    https://
-    (?:bandcamp[.])?
-    (?P<label>(?!bandcamp|com)[^.]+)
-    [.](?!bcbits)[\w/.-]+
-)""",
-        re.VERBOSE,
-    ),
+]
+URL_PATTERNS = [
+    re.compile(r"(?P<url>https://(?P<label>(?!bandcamp\.)[^.]+)\.(?!bcbits)[\w/.-]+)"),  # label-first pattern
+    re.compile(r"(?P<url>https://bandcamp\.(?P<label>[^.]+)\.(?!bcbits)[\w/.-]+)"),  # label-second pattern
```

Comment:
I wonder whether these two patterns could somehow be combined into one? I imagine the presence of named groups is in the way if we wanted to have something like this, right? 
```regex
r"""
(?P<url>https://
    (
          (?P<label>(?!bandcamp\.)[^.]+)
        | bandcamp\.(?P<label>[^.]+)
    )
\.(?!bcbits)[\w/.-]+)
"""
```

Reply:
Unfortunately it's not. The only way to write a single-line regex that accommodates both cases is to have the named group in two places. Even if you declare alterneration between them so that only one can exist per string, Python still won't allow it. A named group can only exist once in a regex, regardless of how many times it can actually be filled.